Main.hs

```diff
@@ -0,0 +1,198 @@
+module Main where
+
+import Control.Monad (unless)
+import Text.Printf (printf)
+
+short :: [a] -> Bool
+short []        = True
+short [x]       = True
+short [x, y]    = True
+short [x, y, z] = False
+short (x:y:z:_) = False
```

Comment:
It's better to use wildcards when you are not using the names introduced in a pattern match.